lib/io.c

```diff
@@ -33,7 +33,8 @@ void metal_io_init(struct metal_io_region *io, void *virt,
 int metal_io_block_read(struct metal_io_region *io, unsigned long offset,
 	       void *restrict dst, int len)
 {
-	void *ptr = metal_io_virt(io, offset);
+	char *ptr = metal_io_virt(io, offset);
```

Comment:
unsigned char is also used in this function, please use char or unsigned char consistently inside the funciton